db/columns.py

```diff
@@ -284,15 +309,50 @@ def get_mathesar_column_with_engine(col, engine):
     return new_column
 
 
-def drop_column(
-        engine,
-        table_oid,
-        column_index,
-):
+def drop_column(table_oid, column_index, engine):
     column_index = int(column_index)
     table = tables.reflect_table_from_oid(table_oid, engine)
     column = table.columns[column_index]
     with engine.begin() as conn:
         ctx = MigrationContext.configure(conn)
         op = Operations(ctx)
         op.drop_column(table.name, column.name, schema=table.schema)
+
+
+def get_column_default(table_oid, column_index, engine):
+    table = tables.reflect_table_from_oid(table_oid, engine)
+    column = table.columns[column_index]
+    return _decode_server_default(column.server_default, engine)
+
+
+def _decode_server_default(server_default, engine):
+    if server_default is not None:
+        cast_sql_text = server_default.arg.text
+        # Ensure that we never evaluate something that increments a query
+        # This is safe as strings will always start with a single quote
+        if cast_sql_text.startswith("nextval(") and cast_sql_text.endswith(")"):
+            return None
```

Comment:
Ooof, this is kinda hard to parse at a glance. I wonder if you could put 'nextval(' in a variable and ')' like:

```suggestion

    INC_TYPE_START='nextval('
    INC_TYPE_END=')'
    is_incrementing_type = cast_sql_text.startswith(INC_TYPE_START) and cast_sql_text.endswith(INC_TYPE_END)
    
    if is_incrementing_type:
       return None
```